src/cpp/core/include/core/r_util/RSourceIndex.hpp

```diff
@@ -253,16 +259,17 @@ class RSourceIndex : boost::noncopyable
                                     _1,
                                     patternRegex,
                                     prefixOnly,
-                                    caseSensitive);
+                                    caseSensitive, 
+                                    includeTestItems);
       }
       else
       {
          if (prefixOnly)
             predicate = boost::bind(&RSourceItem::nameStartsWith,
-                                       _1, term, caseSensitive);
+                                       _1, term, caseSensitive, includeTestItems);
          else
             predicate = boost::bind(&RSourceItem::nameIsSubsequence,
-                                       _1, term, caseSensitive);
+                                       _1, term, caseSensitive, includeTestItems);
       }
 
       return search(newContext, predicate, out);
```

Comment:
What about something like:

```
// create a filter on top of our chosen predicate, based on the appropriate
// set of RSourceItems to be searched
auto filter = [](const RSourceItem& item) {
  // check for tests

  // call selected predicate
  return predicate(item);
}
```

and then allow that filter to be passed along to `search()`?

My main goal is to have a way of separating the "filter based on the type of the RSourceItem" and "filter based on whether the text-content of the source item matches the search term".

Reply:
(let me know if you feel it's not worth the effort; however, it feels like separating the responsibilities here would make it easier to extend in the future)